cartesi-rollups/development/assets-handling/overview.md

```diff
@@ -0,0 +1,51 @@
+---
+id: overview
+title: Overview
+tags: [assets, portals, vouchers]
+resources:
+   - url: https://www.udemy.com/course/the-cartesi-dapp-developer-masterclass
+     title: The Cartesi dApp Developer Free Course
+   - url: https://github.com/jjhbk/cartesi-wallet
+     title: "Cartesi-Wallet: TypeScript-based Wallet Implementation for Cartesi dApps"
+   - url: https://github.com/jjhbk/cartesi-router
+     title: "Cartesi-Router: TypeScript-based Router Implementation for Cartesi dApps"
+   - url: https://github.com/Mugen-Builders/frontend-cartesi-wallet-x
+     title: "Frontend-Cartesi-Wallet: A React frontend to showcase wallet functionality of Cartesi dApps" 
+---
+
+Assets exist on the base layer, where they have actual meaning and value.
+
+As with any execution layer solution, a Cartesi dApp that wants to manipulate assets needs a secure way of "teleporting" the assets from the base layer to the execution layer and then a way to "teleport" them back to the base layer.
+
+Currently, Cartesi Rollups support the following types of assets:
+
+- [Ether (ETH)](../../api/json-rpc/portals/EtherPortal.md)
+- [ERC-20](../../api/json-rpc/portals/ERC20Portal.md)
+- [ERC-721](../../api/json-rpc/portals/ERC721Portal.md)
+- [ERC-1155 Single](../../api/json-rpc/portals/ERC1155SinglePortal.md)
+- [ERC-1155 Batch](../../api/json-rpc/portals/ERC1155BatchPortal.md)
+
+
+![img](../../../static/img/v1.3/assets.jpg)
+
+
+The Portal contracts, part of the on-chain components, are responsible for teleporting assets from the base layer blockchain to a Cartesi dApp. You can think of a Portal as a bank account owned by the off-chain machine.
+
+Once deposited, those L1 assets gain a representation in the execution layer and are owned by whomever the depositor assigned them to.
+
+When an asset is deposited, the corresponding Portal contract sends an input via the InputBox contract, describing the type of asset, amount, receivers, and some data the depositor might want the dApp to read. This allows deposits and instructions to be sent as a single base layer interaction.
+
+Anyone can deposit assets there, but only the dApp — through the [`CartesiDApp`](../../api/json-rpc/application.md) contract — can decide on withdrawals. An input is sent requesting a withdrawal, which gets processed and interpreted off-chain. If everything is correct, the machine creates a voucher destined to the appropriate Portal contract, ordering and finalizing that withdrawal request.
```

Comment:
> InputBox contract, describing the type of asset, amount, **receivers**, and some data the depositor

There are no receivers as part of the deposit on portals. The sender will receive it on the other side. I should list a few fields here later.

Reply:
@jplgarcia the idea is correct. CartesiDapp executes the voucher on itself and ultimately releases the funds.

Reply:
> If everything is correct, the machine creates a voucher destined to the appropriate Portal contract

This is not true. The Cartesi Machine running the dapp code creates a voucher that can be executed when an epoch has settled. The execution happens at the `CartesiDapp` contract, which can verify if the voucher has the correct (merkle) proof.

Reply:
Here are all the function signatures for portals, so you can choose how to better describe in the text.
```solidity
function depositBatchERC1155Token(
        IERC1155 _token,
        address _dapp,
        uint256[] calldata _tokenIds,
        uint256[] calldata _values,
        bytes calldata _baseLayerData,
        bytes calldata _execLayerData
    );
function depositSingleERC1155Token(
        IERC1155 _token,
        address _dapp,
        uint256 _tokenId,
        uint256 _value,
        bytes calldata _baseLayerData,
        bytes calldata _execLayerData
    );
function depositERC20Tokens(
        IERC20 _token,
        address _dapp,
        uint256 _amount,
        bytes calldata _execLayerData
    );  
function depositERC721Token(
        IERC721 _token,
        address _dapp,
        uint256 _tokenId,
        bytes calldata _baseLayerData,
        bytes calldata _execLayerData
    )
function depositEther(
        address _dapp,
        bytes calldata _execLayerData
    ) external payable;
```
The only that is not sent is `address _dapp` since that's is a selector meant for the InputBox itself.